File: Chapter 6/Chapter6.java
interface Shape {
	public double area();
  public double circ();
}

public class Circle 
  implements Shape {
  private double radius;
  
  public Circle(double r){
    radius = r;
  }
  
  public double area() {
    double area = Math.PI * radius * radius;
    return area;
  }

  public double circ() {
    double circumference = 2.0 * Math.PI * radius;
    return circumference;
  }
}

public class Rect 
  implements Shape {
  private double width;
  private double height;
  
  public Rect(double w, double h){
  	width = w;
    height = h;
  }
  
  public double area() {
    double area = width * height;
    return area;
  }
  
  public double circ() {
    double circumference = 2.0 * (width + height);
    return circumference;
  }
}

public class Square 
  implements Shape {
  private double edge;
  
  public Square(double e){
  	edge = e;
  }
  
  public double area() {
    double area = edge * edge;
    return area;
  }
  
  public double circ() {
    double circumference = 2.0 * (edge + edge);
    return circumference;
  }
}

Rect testRect = new Rect(2,2);
System.out.println(testRect.area());
System.out.println(testRect.circ());

File: Chapter 6/Chapter6.hs
module Chapter6 where

-- Exercise 4.

data Shape
  = Circle Float
  | Rect Float Float
  | Square Float

area :: Shape -> Float
area (Circle r) = pi * r * r
area (Rect d h) = d * h
area (Square e) = e * e

circ :: Shape -> Float
circ (Circle r) = 2.0 * pi * r
circ (Rect d h) = 2.0 * (d + h)
circ (Square e) = 2.0 * (e + e)